ElevenLabs/Kernel/ElevenLabs.m

```diff
@@ -0,0 +1,201 @@
+Begin["Wolfram`ElevenLabsAPI`"] (* Begin Private Context *)
+
+
+Begin["`Private`"](* Begin Private Context *)
+
+(******************************* ElevenLabs *************************************)
+
+(* Authentication information *)
+
+ElevenLabsData[] = {
+    "ServiceName" -> "ElevenLabs",
+    "URLFetchFun" :> Function[
+        Enclose @ With[{params = Lookup[{##2}, "Parameters", {}]},
+            HTTPRequest[
+                #1,
+                Association @ FilterRules[{##2}, Except["Parameters" | "AudioStream"]] //
+                    MapAt[Append["xi-api-key" -> Confirm @ Lookup[params, "key"]], "Headers"] //
+                    KeyMap[Replace["BodyData" -> "Body"]] //
+                    MapAt[DeleteCases[None], "Body"]
+            ] // If[Lookup[params, "AudioStream", False],
+                (* bad attempt at supporting audio streaming *)
+                With[{file = CreateFile[]},
+                    URLDownloadSubmit[#, file]; While[FileByteCount[file] == 0]; Audio[file]
```

Comment:
The code comment suggests this was just meant to be a quick solution, but I think it's probably worth trying to avoid having an infinite busy loop here. What if this was instead:

```wolfram
task = URLDownloadSubmit[#, file];
TaskWait[task];
Audio[file]
```

This still needs error handling etc. obviously, but at least it won't consume CPU cycles.

Reply:
This still won't be satisfactory. It would be nice to return an AudioStream, which can play immediately, even before the full speech is generated. Right now, it returns an incomplete Audio tied to a remote file, but it requires remaking it to reload its size.